src/chunk_append/exec.c

```diff
@@ -648,7 +634,12 @@ chunk_append_initialize_dsm(CustomScanState *node, ParallelContext *pcxt, void *
 	state->lock = chunk_append_get_lock_pointer();
 	pstate->next_plan = INVALID_SUBPLAN_INDEX;
 
-	state->choose_next_subplan = choose_next_subplan_for_leader;
+	/*
+	 * Leader should use the same subplan selection as normal worker threads. If the user wishes to
+	 * disallow running plans on the leader they should do so via the parallel_leader_participation
+	 * GUC.
+	 */
+	state->choose_next_subplan = choose_next_subplan_for_worker;
 	state->current = INVALID_SUBPLAN_INDEX;
 	state->pcxt = pcxt;
 	state->pstate = pstate;
```

Comment:
This is now duplicated code in both `chunk_append_initialize_dsm` and `chunk_append_initialize_worker`. It would be a good idea to move this out of the functions and into the caller, or at least factor it out into a separate function.

Reply:
I'm a bit reluctant to refactor this too much as there is a subtle interaction between these functions (initialize_worker uses the parallel state set up by initialize_dsm).  There's really only a couple lines that can be trivially refactored here.